### ISSUE TITLE
[Q] Possible to specify an on-premise S3 storage with intake?

### ISSUE THREAD

**Issue by cwerner** [intake/intake#518]
Hi, new to intake so forgive me if this is mentioned somewhere.

Can I specify an on-premise object store that follows the S3 AWS conventions (mini or netapp)? I can use it via boto3 or s3fs in my scripts, just wondering if the config can be specified or if it always defaults to the AWS servers for s3? 

If this is possible - are there any examples that I can look at?

**Comment by martindurant** [comment]
Most of the data source drivers take a "storage_options" parameter, where the values allowed depend on the backend filesystem, in this case s3fs. For a situation like yours, you will likely end up with something like:
```yaml
sources:
  mysource:
    name: ..
    driver: ..
    args:
      storage_options:
        client_kwargs: {"endpoint_url": endpoint_uri}
    ...
```
where you will supply the URL of your internal storage.

botocore also allows you to define ["profiles" ](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-a-configuration-file), which would be included I think just as
```yaml
      storage_options:
        profile_name: myprofile
    ...
```
Basically, whatever you need to make s3fs work can be passed to an intake source which is designed to work with remote sources (which is most of them).

**Comment by cwerner** [comment]
Thanks - very cool!

**Comment by martindurant** [comment]
OK to close?

**Comment by cwerner** [comment]
Yes, sorry for not closing it myself ...
